File: eBotanika.Services/Database/Korisnik.cs
namespace eBotanika.Services.Database
{
    public partial class Korisnik
    {
        public Korisnik()
        {
            Placanje = new HashSet<Placanje>();
            Ocjena = new HashSet<Ocjena>();
            Rezervacije = new HashSet<Rezervacije>();
        }

        public int KorisnikID { get; set; }
        public string Ime { get; set; }
        public string Prezime { get; set; }
        public DateTime DatumRodjenja { get; set; }
        public string Email { get; set; }
        public string Telefon { get; set; }
        public string KorisnickoIme { get; set; }
        public string LozinkaHash { get; set; }
        public string LozinkaSalt { get; set; }
        public ICollection<Placanje> Placanje { get; set; }
        public ICollection<Rezervacije> Rezervacije { get; set; }
        public ICollection<Ocjena> Ocjena { get; set; }
    }
}

File: eBotanika.Mobile/lib/models/korisnik.dart
class Korisnik {
  int korisnikID;
  String ime;
  String prezime;
  DateTime datumRodjenja;
  String email;
  int telefon;
  String korisnickoIme;
  String lozinka;
  String lozinkaPotvrda;

  Korisnik(
      {required this.korisnikID,
      required this.ime,
      required this.prezime,
      required this.datumRodjenja,
      required this.email,
      required this.telefon,
      required this.korisnickoIme,
      required this.lozinka,
      required this.lozinkaPotvrda});

  factory Korisnik.fromJson(Map<String, dynamic> json) {
    return Korisnik(
        korisnikID: int.parse(json["korisnikID"].toString()),
        ime: json["ime"],
        prezime: json["prezime"],
        datumRodjenja: json["datumRodjenja"],
        email: json["email"],
        telefon: json["telefon"],
        korisnickoIme: json["korisnickoIme"],
        lozinka: json["lozinka"],
        lozinkaPotvrda: json["lozinkaPotvrda"]);
  }

  Map<String, dynamic> toJson() => {
        "korisnikID": korisnikID,
        "ime": ime,
        "prezime": prezime,
        "datumRodjenja": datumRodjenja,
        "email": email,
        "telefon": telefon,
        "korisnickoIme": korisnickoIme,
        "lozinka": lozinka,
        "lozinkaPotvrda": lozinkaPotvrda
      };
}
